### ISSUE TITLE
Multilabel Classification (no softmax) with train.py

### ISSUE THREAD

**Issue by AMelatur** [MaximIntegratedAI/ai8x-training#143]
Hello! I would like to ask how to accomplish multi-label (not multi-class) classification using the train.py script. Indeed, I have an image processing task with 14 output classes that are not mutually exclusive, so where multiple labels can be true at the same time (i.e. person smiling, wearing a hat, having long hair, etc.). 

The only flag I saw to change the loss function was "--regression", but nothing else - basically my question is how to remove the softmax on the outputs, and only apply a sigmoid on each one of the 14 output values.

Thank you!

**Comment by AMelatur** [comment]
Hello! I was just wondering if there was any solution to this issue yet.

Thank you!

**Comment by ermanok** [comment]
You can apply the sigmoid function at the end of the forward function of your model. But, in the hardware, we do not support sigmoid, so you have to implement this function in C after `unload`ing the output from the CNN unit's memory.

When you run train.py in classification mode, it uses CrossEntropy loss, so if you want to use other loss functions you have to modify the train.py to call your function.
